### PR TITLE
Added naviage to page function

### PR DESCRIPTION
**Pull request by gaurav-thoughtspot**
https://thoughtspot.atlassian.net/browse/SCAL-94519

### CODE REVIEW

File: src/embed/ts-embed.ts [hunk 1]
```diff
@@ -503,6 +503,11 @@ export class TsEmbed {
         return this;
     }
 
+    public navigateToPage(path: string): void {
+        const iframeSrc = this.iFrame.src;
+        this.iFrame.src = `${iframeSrc.split('#/embed')[0]}#/embed/${path}`;
```
[the remark this answers is not in the document]
Comment (gaurav-thoughtspot) [reply src/embed/ts-embed.ts]: Fixed

File: src/embed/ts-embed.ts [hunk 2]
```diff
@@ -503,6 +503,21 @@ export class TsEmbed {
...
     }
 
+    /**
+     * Navigate to particular page. eg:answers/pinboards/home
+     * This is used for embedding answers, pinboards, visualizations and full application.
+     * @param path The string, set to iframe src and navigate to new page
+     * eg: appEmbed.navigateToPage('pinboards')
+     */
+    public navigateToPage(path: string): void {
+        const iframeSrc = this.iFrame.src;
```
Comment (jbhanu-thoughtspot) [inline src/embed/ts-embed.ts:513]: More simplified:
```
const iframeSrc = this.iFrame.src;
const embedPath = '#/embed';
const currentPath = iframeSrc.includes(embedPath) ? embedPath : '#';
this.iFrame.src = `${iframeSrc.split(currentPath)[0]}${currentPath}/${path}`;
```

File: src/embed/ts-embed.ts [hunk 3]
```diff
@@ -503,6 +503,21 @@ export class TsEmbed {
         return this;
     }
 
+    /**
+     * Navigate to particular page. eg:answers/pinboards/home
+     * This is used for embedding answers, pinboards, visualizations and full application.
+     * @param path The string, set to iframe src and navigate to new page
+     * eg: appEmbed.navigateToPage('pinboards')
+     */
+    public navigateToPage(path: string): void {
```
[the remark this answers is not in the document]
Comment (gaurav-thoughtspot) [reply src/embed/ts-embed.ts:515]: Done

File: src/embed/ts-embed.ts (hunk 2 above)
Comment (gaurav-thoughtspot) [reply src/embed/ts-embed.ts:516]: updated

File: src/embed/ts-embed.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jbhanu-thoughtspot) [reply src/embed/ts-embed.ts:515]: Don't see any change related to this?

File: src/embed/ts-embed.ts [hunk 4]
```diff
@@ -503,6 +503,24 @@ export class TsEmbed {
         return this;
     }
 
+    /**
+     * Navigate to particular page. eg:answers/pinboards/home
+     * This is used for embedding answers, pinboards, visualizations and full application.
+     * @param path The string, set to iframe src and navigate to new page
+     * eg: appEmbed.navigateToPage('pinboards')
+     * and also strip off any existing / in the path input.
```
Comment (gaurav-thoughtspot) [inline src/embed/ts-embed.ts:511]: Here is change

File: src/embed/ts-embed.ts [hunk 5]
```diff
@@ -503,6 +503,24 @@ export class TsEmbed {
...
+        eg: path = #/pinboards
+        or path = /pinboards
+     */
+    public navigateToPage(path: string): void {
+        const iframeSrc = this.iFrame.src;
+        const embedPath = '#/embed';
```
Comment (gaurav-thoughtspot) [inline src/embed/ts-embed.ts:517]: Here is change

File: src/embed/ts-embed.ts (hunk 4 above)
Comment (jbhanu-thoughtspot) [reply src/embed/ts-embed.ts:511]: Let's handle it inside the code as well, if the path starts with `#/` or `/`  we need to remove that part.
path = path.replace(/^\/?#?\//, '');

**Review by jbhanu-thoughtspot — APPROVED** [review]
lgtm

**Pull request merged**